roles/dnsmasq/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+---
+- name: install dnsmasq and bind-utils
+  sudo: yes
+  yum:
+    name: "{{ item }}"
+    state: latest
+  with_items:
+    - dnsmasq
+    - bind-utils
+  when: inventory_hostname in groups[master_group_name]
```

Comment:
Does it really some reason to put dnsmasq on every node? HA? Local availability?